src/plugin.js

```diff
@@ -695,46 +789,91 @@ void main() {
-      window.navigator.getVRDisplays().then((displays) => {
-        if (displays.length > 0) {
-          this.log('Displays found', displays);
-          this.vrDisplay = displays[0];
-
-          // Native WebVR Head Mounted Displays (HMDs) like the HTC Vive
-          // also need the cardboard button to enter fully immersive mode
-          // so, we want to add the button if we're not polyfilled.
-          if (!this.vrDisplay.isPolyfilled) {
-            this.log('Real HMD found using VRControls', this.vrDisplay);
-            this.addCardboardButton_();
-
-            // We use VRControls here since we are working with an HMD
-            // and we only want orientation controls.
-            this.controls3d = new VRControls(this.camera);
-          }
-        }
 
-        if (!this.controls3d) {
-          this.log('no HMD found Using Orbit & Orientation Controls');
-          const options = {
-            camera: this.camera,
-            canvas: this.renderedCanvas,
-            // check if its a half sphere view projection
-            halfView: this.currentProjection_.indexOf('180') === 0,
-            orientation: videojs.browser.IS_IOS || videojs.browser.IS_ANDROID || false
-          };
-
-          if (this.options_.motionControls === false) {
-            options.orientation = false;
-          }
+      window.navigator.getVRDisplays().then((displaysArray) => {
+        displays = displaysArray;
+      });
+    }
 
-          this.controls3d = new OrbitOrientationContols(options);
-          this.canvasPlayerControls = new CanvasPlayerControls(this.player_, this.renderedCanvas, this.options_);
-        }
+    // Detect WebXR is supported
+    if (window.navigator.xr) {
+      this.log('WebXR is supported');
+      window.navigator.xr.isSessionSupported('immersive-vr').then((supportsImmersiveVR) => {
+        if (supportsImmersiveVR) {
+          // We support WebXR show the enter VRButton
+          this.vrButton = VRButton.createButton(this.renderer);
+          document.body.appendChild(this.vrButton);
+          this.initImmersiveVR();
+          this.initXRPolyfill(displays);
+
+          // key or pointer click toogle playback
+          document.body.addEventListener('keydown', () => {
+            this.togglePlay_();
+          });
+          document.body.addEventListener('click', () => {
+            this.togglePlay_();
+          });
 
-        this.animationFrameId_ = this.requestAnimationFrame(this.animate_);
+        } else {
+          // fallback to older WebVR if WebXR immersive session is not available
+          this.initVRPolyfill(displays);
+        }
+        window.navigator.xr.setSession = (session) => {
+          this.currentSession = session;
+          this.renderer.xr.setSession(this.currentSession);
+        };
       });
+    } else {
+      // fallback to older WebVR if WebXR Device API is not available
+      this.initVRPolyfill(displays);
+    }
```

Comment:
The variable `displays` is set in a promise, I don't see how this could work but maybe I'm missing something.

Reply:
This seems to lead to an NPE on the property `this.player_.vr().vrDisplay` in the following file :https://github.com/mister-ben/videojs-vr/blob/4576704f9c807b632a744607bfed8a084dedf509/src/cardboard-button.js#L42-L49